Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTestEnumerationSuiteFactory.cpp

```diff
@@ -84,4 +87,86 @@ namespace TestImpact
             return testSuites;
         }
     } // namespace GTest
+
+    namespace Python
+    {
+        AZStd::vector<PythonTestEnumerationSuite> TestEnumerationSuitesFactory(
+            const AZStd::string& testEnumerationData)
+        {
+            size_t startLine = 0;
+            size_t endLine = 0;
+
+            AZStd::vector<PythonTestEnumerationSuite> pairs;
+            AZStd::map<AZStd::string, AZStd::vector<TestEnumerationSuite>> testSuiteMap;
+            AZStd::string previousTestFixture;
+
+            while ((startLine = testEnumerationData.find_first_not_of('\n', endLine)) != AZStd::string::npos)
+            {
+                enum MatchEntries
+                {
+                    FullMatch,
+                    ScriptPath,
+                    TestFixture,
+                    TestName,
+                    NumMatches
+                };
+
+                endLine = testEnumerationData.find('\n', startLine);
+                AZStd::string curLine = testEnumerationData.substr(startLine, endLine - startLine);
+
+                // This regex matches pytest test names, in the form ScriptPath::TestFixture::TestName. It will accept both parameterized
+                // and unparamaterized tests (with or without [] at the end).
+                const AZStd::basic_regex testNamePattern = AZStd::basic_regex("(.*py)::([A-Za-z_/\\-0-9]*)::(.*)");
+
+                // Steps through our string line by line. Assumes lines are split by newline characters.
+                AZStd::smatch matchResults;
+                if (AZStd::regex_search(curLine, matchResults, testNamePattern))
+                {
+                    AZ_TestImpact_Eval(
+                        matchResults.size() == NumMatches,
+                        ArtifactException,
+                        "Error, TestSuite match results did not equal expected number of capture groups");
+                    AZStd::string absoluteScriptPath = matchResults[ScriptPath];
+                    AZStd::string testFixture = matchResults[TestFixture];
+                    AZStd::string testName = matchResults[TestName];
+
+                    // Fetch or create the vector for this script
+                    AZStd::vector<TestEnumerationSuite>& suitesForThisScript = testSuiteMap[absoluteScriptPath];
+
+                    // If we're not working with the same fixture as our prevous iteration, or our suites vector is empty, create a test
+                    // suite, add the current test case to it, and add the suite to our vector.
```

Comment:
Uber nit: the TIAF code only uses full stops for API comments (check other comments as well).